### PR TITLE
[flang][openacc] Lower update directive

### PR DESCRIPTION
**Pull request by clementval**

### CODE REVIEW

File: flang/lib/Lower/OpenACC.cpp [hunk 1]
```diff
@@ -516,6 +528,110 @@ genACC(Fortran::lower::AbstractConverter &converter,
...
+                   std::get_if<Fortran::parser::AccClause::Host>(&clause.u)) {
+      genObjectList(hostClause->v, converter, hostOperands);
+    } else if (const auto *deviceClause =
+                   std::get_if<Fortran::parser::AccClause::Device>(&clause.u)) {
+      genObjectList(deviceClause->v, converter, deviceOperands);
+    }
+  }
+
+  // Prepare the operand segement size attribute and the operands value range.
+  SmallVector<Value, 8> operands;
```
Comment (schweitzpgi) [inline flang/lib/Lower/OpenACC.cpp:592]: `Value` can be ambiguous type.

File: flang/lib/Lower/OpenACC.cpp [hunk 2]
```diff
@@ -516,6 +528,110 @@ genACC(Fortran::lower::AbstractConverter &converter,
...
+      }
+    } else if (const auto *waitClause =
+                   std::get_if<Fortran::parser::AccClause::Wait>(&clause.u)) {
+      const auto &waitClauseValue = waitClause->v;
+      if (waitClauseValue) { // wait has a value.
+        const Fortran::parser::AccWaitArgument &waitArg = *waitClauseValue;
+        const std::list<Fortran::parser::ScalarIntExpr> &waitList =
+            std::get<std::list<Fortran::parser::ScalarIntExpr>>(waitArg.t);
+        for (const Fortran::parser::ScalarIntExpr &value : waitList) {
+          Value v = fir::getBase(
```
Comment (schweitzpgi) [inline flang/lib/Lower/OpenACC.cpp:569]: Value

**Review by schweitzpgi — APPROVED** [review]
Looks like these PRs generated some conflicts.

**Comment by clementval** [comment]
> Looks like these PRs generated some conflicts.

Yeah ... It was not a good idea in the end to issue split PRs. I'll have to rebase after each merge.

**Comment by clementval** [comment]
@schweitzpgi Conflict have been resolved. Might have new conflict if you merge #526 before.

**Pull request merged**